raiden/network/transport/matrix/transport.py

```diff
@@ -470,6 +471,24 @@ def on_success(greenlet):
         self.greenlets.append(greenlet)
         return greenlet
 
+    def _schedule(self, period: float, func: Callable, *args, **kwargs):
+        """Spawns two greenlets: one to call func with args immediately,
+        another one to call this function again after period has passed."""
+
+        def on_success(greenlet):
+            if greenlet in self.greenlets:
+                self.greenlets.remove(greenlet)
+
+        run_greenlet = gevent.spawn_later(0, func, *args, **kwargs)
+        run_greenlet.link_exception(self.on_error)
+        run_greenlet.link_value(on_success)
+
+        schedeule_greenlet = gevent.spawn_later(
```

Comment:
Not sure I can follow, could you elaborate please?

 For me this approach has the benefit, that the executed function itself just takes the time it takes and the scheduling takes place in the event loop, depending on how busy it currently is.
 I had a while loop with sleep in it first, it produced timeout errors in many tests, as the `greenlets.get()` in `transport.stop()` was waiting for the sleep to finish. 
Also, why 15 seconds? I think this depends on how busy the main loop is, which makes sense to me.

Reply:
15s because on average the stop event will happen halfway through the 30s sleep (law of large numbers...)

Instead of a simple sleep you could use `gevent.wait(stop_event, 30)`. That way it will immediately return once the stop event is set.

Reply:
I misunderstood your first comment. 
The case you described is handled by transport.scheduled_greenlets.killall() in transport.stop() s.t. the parent does not have to wait for the scheduled children. I prefer the spawn_later approach, as we don‘t have to write an explicit while loop for every singular greenlet that’s supposed to run periodically. If we want to have greenlets that are not supposed to be killed on flight, there sure is a way to only kill those scheduled for later execution.

Reply:
The question is will we ever have more than this one periodic thing?

Smell a bit [YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) to me ;)

Reply:
For instance [#3262](https://github.com/raiden-network/raiden/issues/3262) is something that should be done/checked periodically i think